lib/internal/modules/run_main.js

```diff
@@ -33,8 +34,11 @@ function shouldUseESMLoader(mainPath) {
     return true;
   if (!mainPath || mainPath.endsWith('.cjs'))
     return false;
-  const pkg = readPackageScope(mainPath);
-  return pkg && pkg.data.type === 'module';
+  const pkgType = readPackageScope(mainPath)?.data?.type ?? 'none';
+  return pkgType === 'module' || (
+    pkgType === 'none' &&
+    defaultPackageType === 'module'
+  );
```

Comment:
Can we just make the default `"commonjs"` instead of `"none"` since that is what we've defined to be the default.